azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_servicefabric.go

```diff
@@ -0,0 +1,269 @@
+func ResourceServiceEndpointServiceFabric() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointServiceFabric, expandServiceEndpointServiceFabric)
+
+	r.Schema["cluster_endpoint"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Client connection endpoint for the cluster. Prefix the value with 'tcp://';. This value overrides the publish profile.",
+	}
+
+	secretHashKeyClientCertificate, secretHashSchemaClientCertificate := tfhelper.GenerateSecreteMemoSchema("client_certificate")
+	secretHashKeyClientCertificatePassword, secretHashSchemaClientCertificatePassword := tfhelper.GenerateSecreteMemoSchema("client_certificate_password")
+	r.Schema[resourceBlockServiceFabricCertificate] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"server_certificate_lookup":      servicefabricServerCertificateLookupSchema(),
+				"server_certificate_thumbprint":  servicefabricServerCertificateThumbprintSchema(resourceBlockServiceFabricCertificate),
+				"server_certificate_common_name": servicefabricServerCertificateCommonNameSchema(resourceBlockServiceFabricCertificate),
+				"client_certificate": {
+					Type:             schema.TypeString,
+					Required:         true,
+					Description:      "Base64 encoding of the cluster's client certificate file.",
+					Sensitive:        true,
+					DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+				},
+				"client_certificate_password": {
+					Type:             schema.TypeString,
+					Required:         true,
+					Description:      "Password for the certificate.",
+					Sensitive:        true,
+					DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+				},
+				secretHashKeyClientCertificate:         secretHashSchemaClientCertificate,
+				secretHashKeyClientCertificatePassword: secretHashSchemaClientCertificatePassword,
+			},
+		},
+		ConflictsWith: []string{resourceBlockServiceFabricAzureActiveDirectory, resourceBlockServiceFabricNone},
+	}
+
+	secretHashKeyPassword, secretHashSchemaPassword := tfhelper.GenerateSecreteMemoSchema("password")
+	r.Schema[resourceBlockServiceFabricAzureActiveDirectory] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"server_certificate_lookup":      servicefabricServerCertificateLookupSchema(),
+				"server_certificate_thumbprint":  servicefabricServerCertificateThumbprintSchema(resourceBlockServiceFabricAzureActiveDirectory),
+				"server_certificate_common_name": servicefabricServerCertificateCommonNameSchema(resourceBlockServiceFabricAzureActiveDirectory),
+				"username": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Specify an Azure Active Directory account.",
```

Comment:
Should add a valitaion: `ValidateFunc: validation.StringIsNotEmpty,`  for `username`